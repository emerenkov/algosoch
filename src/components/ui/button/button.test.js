import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

//use snapshots for correct test
describe('test component Button', () => {
    it('button has text', () => {
        //render button with text
        const tree = renderer.create(<Button text={'test text'} />).toJSON();
        //test toMatchSnapshot() between in data & out data
        expect(tree).toMatchSnapshot();
    });

    it('button has not text', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('block button', () => {
        const tree = renderer.create(<Button disabled />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('button has loading icon', () => {
        const tree = renderer.create(<Button isLoader />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('press on button appear alert', () => {
        window.alert = jest.fn();
        render(
            <Button onClick={() => window.alert('text')}
                    text={'test text'}
            />)
        const button = screen.getByText('test text');
        //fake press button
        fireEvent.click(button);
        //test works alert
        expect(window.alert).toHaveBeenCalledWith('text');
    })
});
