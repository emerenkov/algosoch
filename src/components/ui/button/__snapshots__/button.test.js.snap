// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component Button block button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`test component Button button has loading icon 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`test component Button button has not text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`test component Button button has text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test text
  </p>
</button>
`;
